### ISSUE TITLE
Backport the test framework to 2.x branch

### ISSUE THREAD

**Issue by DarshitChanpura** [opensearch-project/security#3230]
As part of creating a new test framework, few PRs were added to main but have not been backported to 2.x yet. As part of this effort, these need to be backported so any features that require to be added to 2.x will not be blocked due to missing test framework.
Here are the PRs that need to be backported:

- [ ] https://github.com/opensearch-project/security/pull/1967
- [ ] https://github.com/opensearch-project/security/pull/2138
- [ ] https://github.com/opensearch-project/security/pull/2139
- [ ] https://github.com/opensearch-project/security/pull/2153
- [ ] https://github.com/opensearch-project/security/pull/2155
- [ ] https://github.com/opensearch-project/security/pull/2245

Once these backports are complete, PRs like this https://github.com/opensearch-project/security/pull/3192 will be unblocked from getting merged into 2.x

**Comment by scrawfor99** [comment]
This is required for backporting/releasing OBO/Service Accounts & Auth in Rest Layer.

**Comment by davidlago** [comment]
Changes merged, closing
